language-server/src/server.js

```diff
@@ -305,4 +310,25 @@ connection.languages.semanticTokens.onDelta(({ textDocument, previousResultId })
   return builder.buildEdits();
 });
 
+// $SCHEMA COMPLETION
+connection.onCompletion((textDocumentPosition) => {
+  connection.console.log("called");
+  const doc = documents.get(textDocumentPosition.textDocument.uri);
+  if (!doc) {
+    return [];
+  }
+
+  const instance = JsoncInstance.fromTextDocument(doc);
+  const currentProperty = instance.findProperty(instance);
+
+  if (currentProperty?.key?.value === "$schema" && currentProperty?.value?.value === "") {
+    return getDialectIds().map((uri) => {
```

Comment:
Try to use [match-sorter](https://www.npmjs.com/package/match-sorter) or something similar here. The idea is to filter and sort based on what the user types to make it easier for users to select the suggestion they want.